cmd/supernode/app/root.go

```diff
@@ -65,121 +121,211 @@ func setupFlags(cmd *cobra.Command, opt *Options) {
 	// when this action is called directly.
 	flagSet := cmd.Flags()
 
-	flagSet.StringVar(&configFilePath, "config", configFilePath,
+	defaultBaseProperties := config.NewBaseProperties()
+
+	flagSet.String("config", config.DefaultSupernodeConfigFilePath,
 		"the path of supernode's configuration file")
 
-	flagSet.IntVar(&opt.ListenPort, "port", opt.ListenPort,
+	flagSet.Int("port", defaultBaseProperties.ListenPort,
 		"listenPort is the port that supernode server listens on")
 
-	flagSet.IntVar(&opt.DownloadPort, "download-port", opt.DownloadPort,
+	flagSet.Int("download-port", defaultBaseProperties.DownloadPort,
 		"downloadPort is the port for download files from supernode")
 
-	flagSet.StringVar(&opt.HomeDir, "home-dir", opt.HomeDir,
+	flagSet.String("home-dir", defaultBaseProperties.HomeDir,
 		"homeDir is the working directory of supernode")
 
-	flagSet.StringVar(&opt.DownloadPath, "download-path", opt.DownloadPath,
+	flagSet.String("download-path", defaultBaseProperties.DownloadPath,
```

Comment:
In fact, if you change the `HomeDir`, the `DownloadPath` should follow the update in default. And you can fix that [here](https://github.com/dragonflyoss/Dragonfly/blob/master/supernode/config/config.go#L96).

Reply:
Done!